packages/block/src/from-rpc.ts

```diff
@@ -32,19 +33,30 @@ export default function blockFromRpc(
   if (blockParams.transactions) {
     for (const _txParams of blockParams.transactions) {
       const txParams = normalizeTxParams(_txParams)
+
       // override from address
-      const fromAddress = toBuffer(txParams.from)
+      const fromAddress = new Address(toBuffer(txParams.from))
       delete txParams.from
 
-      const tx = new FakeTransaction(txParams, chainOptions as TransactionOptions)
-      tx.from = fromAddress
+      // get common based on block's number
+      // since hardfork param may not be set
+      // (see TODO in block.constructor)
+      const blockNumber = parseInt(block.header.number.toString('hex'), 16)
```

Comment:
As this is done multiple times, maybe it should belong to `Common`?

Reply:
oh right, sure, I can do it as a static factory `Common.forBlockNumber(chain, blockNumber): Common` 

I also considered something like a helper `Common.hardforkForBlockNumber(chain, blockNumber): string` but I think the static factory is a nice api pattern that we can continue to use. And if really needed you could just access the hardfork with `common.hardfork()`.

Reply:
added in https://github.com/ethereumjs/ethereumjs-vm/pull/812/commits/e425377791fe3b5da750bbda520077ab27ff8567

Reply:
I may remove this in favor of `Common.setHardforkByBlockNumber()` which is being added in https://github.com/ethereumjs/ethereumjs-vm/pull/863.